spec/requests/api/v1/pages_spec.rb

```diff
@@ -20,6 +20,75 @@
     end
   end
 
+  context "when unauthenticated and post a new page" do
+    it "returns unauthorized" do
+      post api_pages_path, params: { title: "Doesn't Matter" }.to_json, headers: v1_headers
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  context "when unauthenticated and update a page" do
+    it "returns unauthorized" do
+      put api_page_path(page.id), params: page.attributes.merge(title: "Doesn't Matter").to_json, headers: v1_headers
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  context "when unauthenticated and delete a page" do
+    it "returns unauthorized" do
+      delete api_page_path(page.id), headers: v1_headers
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  context "when authenticated but not authorized to edit pages" do
+    let(:non_auth_header) do
+      non_admin_secret = create(:api_secret)
+      v1_headers.merge "api-key" => non_admin_secret.secret
+    end
+
+    it "returns unauthorized from create" do
+      post api_pages_path, params: { title: "Also Doesn't Matter" }.to_json, headers: non_auth_header
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+    it "returns unauthorize from update" do
+      put api_page_path(page.id), params: page.attributes.merge(title: "Doesn't Matter").to_json,
+                                  headers: non_auth_header
+      expect(response).to have_http_status(:unauthorized)
+    end
+
+    it "returns unauthorize from destroy" do
+      delete api_page_path(page.id), headers: non_auth_header
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  context "when authenticated and authorized" do
+    include_context "when user is authorized"
+
+    let(:post_params) do
+      attributes_for(:page)
+    end
+
+    it "can create a new page via post" do
+      post api_pages_path, params: post_params.to_json, headers: auth_header
+      expect(response).to have_http_status(:success)
+    end
+
+    it "can update an existing page via put" do
+      put api_page_path(page.id), params: page.attributes.merge(title: "Brand New Title").to_json, headers: auth_header
+      expect(response).to have_http_status(:success)
+      expect(page.reload.title).to eq("Brand New Title")
+    end
+
+    it "can destroy an existing page via delete" do
+      delete api_page_path(page.id), headers: auth_header
+      expect(response).to have_http_status(:success)
+      expect(Page).not_to exist(page.id)
```

Comment:
TIL!

Reply:
Fwiw, this is rubocop's preferred form, but I had originally written it as:

```ruby
expect(Page.exists?(page.id)).to be_false
```